source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -146,6 +146,11 @@ std::string Utility::generalNameAsString(const GENERAL_NAME* general_name) {
     san.assign(reinterpret_cast<const char*>(ASN1_STRING_data(str)), ASN1_STRING_length(str));
     break;
   }
+  case GEN_EMAIL: {
+    ASN1_STRING* str = general_name->d.rfc822Name;
+    san.assign(reinterpret_cast<const char*>(ASN1_STRING_data(str)), ASN1_STRING_length(str));
+    break;
+  }
```

Comment:
Does this have explicit testing somewhere? I don't see any cert changes in this PR?

/wait-any

Reply:
I used this here for testing SAN matcher https://github.com/envoyproxy/envoy/pull/15509/files#diff-ee57e8aee637e17f1f512cef5e2bff807c263ab67922c85659721654913242f4R460, so that's why you don't see cert s changes. 

In any way, `generalNameAsString` doesn't have any explicit unit test so I think we should.